### PR TITLE
tests: test on Linux, macOS and Windows

### PR DESCRIPTION
**Pull request by DanielRuf**
## Description

Test on Linux, macOS and Windows. See https://blog.travis-ci.com/2018-10-11-windows-early-release

## Types of changes

- [ ] Documentation
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (anything that would change an existing functionality)
- [x] Maintenance (refactor, code cleaning, development tools...)

## Checklist

- [x] I have read and follow the CONTRIBUTING.md document.
- [x] The pull request title and template are correctly filled.
- [x] The pull request targets the right branch (`develop` or `develop-v...`).
- [x] My commits are correctly titled and contain all relevant information.
- [ ] I have updated the documentation accordingly to my changes (if relevant).
- [ ] I have added tests to cover my changes (if relevant).

### CODE REVIEW

**Comment by DanielRuf** [comment]
@ncoden not sure if this should be unencrypted https://travis-ci.org/zurb/foundation-sites/jobs/441389318#L441

**Review by ncoden — COMMENTED** [review]
With this new configuration, we are starting tests with:
- Headless Chrome with Node `lts/boron` on windows, linux and macOS
- Headless Chrome with Node `lts/carbon` on windows, linux and macOS
- Headless Chrome with Node `stable` on windows, linux and macOS
- 18 browsers via Browserstack with Node `stable` on windows, linux and macOS

So tests are run 63 times. Are you sure this is what you want to do ?

**Comment by DanielRuf** [comment]
> * 18 browsers via Browserstack with Node `stable` on windows, linux and macOS

This is still WIP.
We ight change this one to run only on one host.

**Comment by ncoden** [comment]
I don't understand how it would be useful (or useful enough to slow tests) to be tested on several versions of Node on Linux, macOS and Window.

We specify the Foundation package as compatible with `node >= 6` so it makes sense to test several versions of Node. But it is up to the Node maintainers to test it on all plateform to ensure it behave the same.

**Comment by ncoden** [comment]
Ok I see. Plateform-specific dependencies (like Sass).

**Comment by ncoden** [comment]
@DanielRuf The `npm` installation is failing on Windows: https://travis-ci.org/zurb/foundation-sites/jobs/441821489#L296

**Comment by DanielRuf** [comment]
I know, this is an issue with nvs and the Windows support is still in Beta.

Also all major Node versions should be checked on all platforms. Time does not matter for the testsuite.

**Comment by DanielRuf** [comment]
Also lts/* does not work with nvs in many cases (was fixed now) but 6, 8 and 10 is better to understand as new LTS will get other names so it makes not much sense to use lts/*.

**Comment by DanielRuf** [comment]
> Also I would prefer to stay with lts versions instead of 8/10

These are LTS branches / releases. Every even release branch will become LTS, every odd one is shortliving (5, 7, 9).

**Comment by DanielRuf** [comment]
6 will use the latest release of Node.js 6. Also keep in mind that this is just for CI which has always the latest branch releases.

**Comment by DanielRuf** [comment]
Also see https://travis-ci.community/t/windows-instances-hanging-before-install/250/9

And it seems the Windows builds leak probably secure env vars.

**Comment by DanielRuf** [comment]
See https://github.com/jasongin/nvs/issues/66 (basically npm is broken on Windows in specific cases).

**Comment by ncoden** [comment]
@DanielRuf Let me know when you gets it working on Window 😉 (and good luck!)

**Comment by ncoden** [comment]
@DanielRuf We could use Travis "Stages" too. https://docs.travis-ci.com/user/build-stages

Like in the first stage `Tests`:
- Unit tests (Linux + Node 6)
- Unit tests (Linux + Node 8)
- Unit tests (Linux + Node 10)

Then if it successes, in a second stage `Additional Tests`:
- BrowserStack (Linux + Node 10)
- Unit tests (macOS + Node 6)
- Unit tests (macOS + Node 8)
- Unit tests (macOS + Node 10)
- Unit tests (Windows + Node 6)
- Unit tests (Windows + Node 8)
- Unit tests (Windows + Node 10)

Travis doens't run more than 3 job in parallel so it would not slow down tests and stop the Travis tests sooner if Unit tests fails. I moved BrowserStack to first position because it takes 7x longer to run (so others tests will run in parrallel).

**Comment by DanielRuf** [comment]
Currently just the puppeteer issue exists (the dll issue), the remaining steps work now.

**Comment by DanielRuf** [comment]
> @DanielRuf What's the status of this?

Still the same, WIP.

**Comment by DanielRuf** [comment]
The Puppeteer dependency is not compatible as you can see in the log of the failing build.

**Comment by ncoden** [comment]
Tests fails because a process continue to run in Windows. It may be Yarn. https://travis-ci.community/t/build-doesnt-finish-after-completing-tests/288/

**Review by ncoden — APPROVED** [review]
@DanielRuf I resolved conflicts, fixed [the Yarn issue that prevented the build to finish](ravis-ci.community/t/build-doesnt-finish-after-completing-tests/288) and added cache for it. LGTM 👍. Say a word and it's merged 😉

File: .travis.yml [hunk 1]
```diff
@@ -1,21 +1,31 @@
...
+os:
+  - windows
+  - linux
+  - osx
+
+node_js:
+  - 6
+  - 8
+  - 10
+  - 11
```
Comment (DanielRuf) [inline .travis.yml]: Let's fail early and use `node` instead of `11` or what do you think @ncoden?

**Pull request merged**